src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/GaussianMixtureModelUnitTest.java

```diff
@@ -0,0 +1,327 @@
+        //Note: sort inferred gaussians because the model is not fully identifiable (up to reordering).
+        //we sort them by the mixture weights
+        final List<MultivariateGaussian> gaussians = new ArrayList<>(gmm.getGaussians());
+        gaussians.sort((g1, g2) -> Double.compare(g1.getpMixtureLog10(), g2.getpMixtureLog10()));
+
+        //now compare inferred results to truth
+        MultivariateGaussian result1 = gaussians.get(0);
+        MultivariateGaussian result2 = gaussians.get(1);
+        assertEquals(gaussians.size(), 2);
+
+        compareWeights(weights, result1, result2, okDistances[0]);
+        compareMeans(mus, result1, result2, okDistances[1]);
+        compareCovariances(sigmas, result1, result2, okDistances[2]);
+    }
+
+    private void compareWeights(double[] weights, MultivariateGaussian result1, MultivariateGaussian result2, double okDistance) {
+        final double[] resultWeigths = {pow(10, result1.getpMixtureLog10()), pow(10, result2.getpMixtureLog10())};
+        for (int i = 0 ; i < resultWeigths.length; i++) {
+            assertTrue(distance(resultWeigths[i], weights[i]) < okDistance, "expected:" + weights[i] + " actual:" + resultWeigths[i]) ;
+        }
+    }
+
+    private void compareMeans(double[][] mus, MultivariateGaussian result1, MultivariateGaussian result2, double okDistance) {
+        final double[][] resultMeans = {result1.getXBar().toArray(), result2.getXBar().toArray()};
+        for (int i = 0 ; i < resultMeans.length; i++) {
+            double dist = distance(resultMeans[i], mus[i]);
+            assertTrue(dist < okDistance, "expected:" + Arrays.toString(mus[i]) + " actual:" + Arrays.toString(resultMeans[i]) + " distance:" + dist);
+        }
+    }
+
+    private void compareCovariances(double[][][] sigmas, MultivariateGaussian result1, MultivariateGaussian result2, double okDistance) {
+        final double[][][] resultCovars = {result1.getParam_S().getData(), result2.getParam_S().getData()};
+        for (int i = 0 ; i < resultCovars.length; i++) {
+            assertTrue(distance(resultCovars[i], sigmas[i]) < okDistance, "expected:" + Arrays.deepToString(sigmas[i]) + " actual:" + Arrays.deepToString(resultCovars[i])) ;
+        }
+    }
+
+    //XXX it is bit lame to compare matrices this way. This returns a max of the abs difference, point-wise.
+    private static double distance(double[][] a, double[][] b) {
+        RealMatrix m1 = new Array2DRowRealMatrix(a);
+        RealMatrix m2 = new Array2DRowRealMatrix(b);
+        RealMatrix m1MinusM2 = m1.subtract(m2);
+        final double[][] diff = m1MinusM2.getData();
+        double max = Double.MIN_VALUE;
+        for(int i = 0; i < diff.length; i++){
+            for(int j = 0; j < diff[i].length; j++){
+                if (abs(diff[i][j]) < max){
+                    max = diff[i][j];
```

Comment:
you are setting the non-absolute value here, but then forgetting to call abs in the assert. Fails to catch half the cases.